### DISCUSSION TITLE
open up to casual contributors

### DISCUSSION THREAD

**Discussion by shinenelson** [tosdr/CrispCMS#54] [Q&A]
I am a person who gives importance to Terms of Service very much. That is also why I know of this project and have been using it for a long time now. However, recently, I found myself reading some Terms myself because they were not annotated by ToS;DR. That was okay for me because I have been reading Terms of Service documents a lot and I know how to navigate them. But another thought came to my mind. 

> What about other people who would probably want to use the service but do not have the time to read through the Terms of Service. 

That is when my _contributing back to the community_ instincts kicked in and I wanted to annotate the terms I was reading. I looked around tosdr.org and could not find any place to submit any service for annotation. Then I realized that the addition of services process was manual and required human intervention. Then I found tosdr/tosdr.org#213 that listed a URL to submit to tosback which I did 3 weeks ago. Until now, those services has not yet been listed for annotation.

So, here is my question : how can a casual contributor ( like me ) who reads Terms of Service documents occasionally be able to contribute that time productively back to ToS;DR?

**Comment by shinenelson** [comment]
a far-fetched goal would be to have an annotator built into the browser extension but I know that would be too much to ask for.

**Comment by JustinBack** [comment]
> Then I found tosdr/tosdr.org#213 that listed a URL to submit to tosback which I did 3 weeks ago. Until now, those services has not yet been listed for annotation.

Because that link is reeeeally old and unused. We have moved over to a service request form @ https://forum.tosdr.org/request

> So, here is my question : how can a casual contributor ( like me ) who reads Terms of Service documents occasionally be able to contribute that time productively back to ToS;DR?

You can always sign up at https://edit.tosdr.org and annotate documents there, everybody can register and contribute, only managing services and points is reserved to a group of selected People :-)

https://forum.tosdr.org/t/how-to-be-a-good-curator-on-tos-dr/84

**Reply by shinenelson** [reply]
> You can always sign up at edit.tosdr.org and annotate documents there, everybody can register and contribute

I already signed up at edit.tosdr.org. But that was not what I was expecting when I opened this discussion. I was looking for a way to annotate documents that I read on the internet ( services that are not already crawled and ready for annotation ) and contribute those annotations back to ToS;DR.

I was reading the [Terms of Service of Hashnode](https://hashnode.com/terms) the other day and wanted to annotate the document. The document had not been crawled before ( I had submitted the document via the _old_ link earlier ). I was reading the document because ToS;DR did not have a rating on the document. And since I was reading the document, I thought I might as well annotate it. That was the means that I was looking for ( the best means would be for the browser extension to enable me to do it - highlight a sentence and choose from the list of options from the context menu option; after all, the annotator is limited to a finite list of options )

I understand that the review of services to be listed is a tedious process and can contain spam / duplicates, etc. But can there be a way for casual 'readers of terms of service and privacy policy documents' annotate documents on the fly as they are reading a document and then submit those annotations back to ToS;DR?

Another impact this could have is for decentralized servers that run services like Mastodon, Diaspora, etc where different servers can have different Terms of Service and one cannot be considered as the same for the others.

Can we open up the process a little more where submissions are more open ( annotations can be submitted by anyone ) but review is more stringent as it is now? So services do not get a class rating unless there are more 'vouches' or something along those lines.

**Reply by JustinBack** [reply]
> I understand that the review of services to be listed is a tedious process and can contain spam / duplicates, etc. But can there be a way for casual 'readers of terms of service and privacy policy documents' annotate documents on the fly as they are reading a document and then submit those annotations back to ToS;DR?

This is definitely possible, but would still require a user account. So you'd have to sign up at edit.tosdr.org anyway.

The idea with the extension annotation is good, though hardly realizable right now due to lack of resources. (Money and time)

Feel free to open an idea discussion in the browser extension repository though, so we can look into this.

https://github.com/tosdr/browser-extensions/discussions/new?category=ideas

I feel once this feature is developed, it will be packed into a different extension as the permissions are probably more funky (Reading website content, current url, etc.)

> Can we open up the process a little more where submissions are more open ( annotations can be submitted by anyone )

I mean, annotation can be submitted by anyone. There is no approval process for this, the only requirement is as you mentioned the account.

Creating services and documents on the go is currently not in mind, as we open a lot of doors for spam. That is why it's a manual process.

I hope I didn't miss anything in your text, if I did, feel free to reach out :-)
